fhir-cql-rest/src/main/java/com/ibm/fhir/cql/engine/rest/retrieve/RestFhirRetrieveProvider.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.ibm.fhir.cql.engine.rest.retrieve;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.core.Response;
+
+import com.ibm.fhir.client.FHIRClient;
+import com.ibm.fhir.client.FHIRParameters;
+import com.ibm.fhir.client.FHIRResponse;
+import com.ibm.fhir.cql.engine.retrieve.SearchParameterFhirRetrieveProvider;
+import com.ibm.fhir.cql.engine.retrieve.SearchParameterMap;
+import com.ibm.fhir.cql.engine.searchparam.IQueryParameter;
+import com.ibm.fhir.cql.engine.searchparam.ReferenceParameter;
+import com.ibm.fhir.cql.engine.searchparam.SearchParameterResolver;
+import com.ibm.fhir.cql.engine.util.FHIRClientUtil;
+import com.ibm.fhir.cql.helpers.FhirBundleCursor;
+import com.ibm.fhir.model.resource.Bundle;
+import com.ibm.fhir.model.resource.Resource;
+import com.ibm.fhir.model.util.ModelSupport;
+
+public class RestFhirRetrieveProvider extends SearchParameterFhirRetrieveProvider {
```

Comment:
Please add some class level comments. Also, if you have Fhir can you use upper case FHIR or not have it in the class name.